lib/manager/poetry/artifacts.ts

```diff
@@ -107,7 +107,8 @@ function getSourceCredentialVars(
   const envVars: Record<string, string> = {};
 
   for (const source of poetrySources) {
-    const matchingHostRule = find({ url: source.url });
+    const matchingHostRule =
+      find({ hostType: 'pypi', url: source.url }) || find({ url: source.url });
     const formattedSourceName = source.name.toUpperCase();
```

Comment:
While we're at it: This is also not completely right. According to [the Poetry documentaion](https://python-poetry.org/docs/master/configuration/#using-environment-variables):

> The environment variables must be prefixed by POETRY_ and are comprised of the uppercase name of the setting and **with dots and dashes replaced by underscore**

Should we also do that here or in a separate PR?

Reply:
This should probably also fixed here, but can also be done in different PR.